python/Register.py

```diff
@@ -6,30 +6,43 @@
 # Always output the following:
 #     Read Data 1 = Registers[Read Addr 1]
 #     Read Data 2 = Registers[Read Addr 2]
-# R0 = 0
+#
 # SP starts at 0xFFFFFFFC
 
 from myhdl import *
 
 @block
 def RegisterFile(BusA, BusB, BusW, RA, RB, RW, RegWr, clk, Rst):
 
-    registers = [intbv(0,0,2**32)] * 32
-
-    @always(clk.posedge)
-    def executePos():
-
-      if RegWr:
-        registers[RW] = BusW
-
-      elif (Rst):
-        for i in range (0,32):
-          registers[i] = intbv(0, 0, 2**32)
-
-    @always(clk.negedge)
-    def executeNeg():
-
-      BusA = registers[RA]
-      BusB = registers[RB]
-
-return executePos, executeNeg
+    registers = []
+    regSig = Signal(0)
+    period = 10
+
+    #Generate Signals of intbv type for registers (Signal is used to track on gtkwave)
+    for i in range(0, 32):
+        registers.append(Signal(intbv(0, 0, 2**32)))
+        registers[i].driven = not registers[i].driven
+
+    #Create clocking iterator to switch every quarter full clock cycle (used to read and write on quarter of clock cycle)
+    @instance
+    def regSig_generator():
+        highTime = int(period/2)
+        lowTime = period - highTime
+
+        while True:
+          yield delay(highTime)
+          regSig.next = 1
+          yield delay(lowTime)
+          regSig.next = 0
```

Comment:
Why not use an instance of Clock_Generator with a different period? You pretty much rewrote the same code.

Thanks for the hard work though in figuring out the issue!

Reply:
I'll go ahead and change that right now. The only reason I did it that way was bc I wanted to be able to mess with when I was setting 0, and 1 in the instance if it like was one step behind the regular clock. It turned out fine and I just never changed it.

Reply:
Sounds good!